apm-agent-plugins/apm-spring-webmvc-plugin/src/test/java/co/elastic/apm/agent/springwebmvc/SpringTransactionNameInstrumentationTest.java

```diff
@@ -79,13 +85,46 @@ void testControllerTransactionName() throws Exception {
         assertThat(reporter.getFirstTransaction().getNameAsString()).isEqualTo("TestController#test");
     }
 
+    @Test
+    void testControllerTransactionNameWhenUsePathAsTransactionNameIsTrue() throws Exception {
+        WebConfiguration webConfiguration = config.getConfig(WebConfiguration.class);
+        when(webConfiguration.isUsePathAsName()).thenReturn(true);
+        mockMvc.perform(get("/test/944207fa-d27b-40f6-be58-6dd6a1b541d4"))
+            .andExpect(content().string("944207fa-d27b-40f6-be58-6dd6a1b541d4"));
+        assertThat(reporter.getFirstTransaction().getNameAsString()).isEqualTo("GET /test/944207fa-d27b-40f6-be58-6dd6a1b541d4");
```

Comment:
We need to minimize the name cardinality, in this case we should expect to have `"/test/{uuid}"` as transaction name, not the one that contains the full path. The full URL path is still captured in transaction attributes so it's not lost and can be retrieved if needed.